controller/appcontroller.go

```diff
@@ -1389,15 +1388,38 @@ func (ctrl *ApplicationController) processAppRefreshQueueItem() (processNext boo
 		localManifests = opState.Operation.Sync.Manifests
 	}
 
-	revision := app.Spec.Source.TargetRevision
-	if comparisonLevel == CompareWithRecent {
-		revision = app.Status.Sync.Revision
-	}
+	revisions := make([]string, 0)
+	sources := make([]appv1.ApplicationSource, 0)
+
+	hasMultipleSources := app.Spec.Sources != nil && len(app.Spec.Sources) > 0
```

Comment:
Noted that you added a HasMultipleSources() method, maybe use that here instead?